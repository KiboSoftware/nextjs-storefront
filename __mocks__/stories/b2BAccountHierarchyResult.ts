export const b2BAccountHierarchyResult = {
  accounts: [
    {
      users: [
        {
          emailAddress: 'sushant.root@gmail.com',
          userName: 'sushant.root@gmail.com',
          firstName: 'Sushant',
          lastName: 'Root Updated',
          localeCode: null,
          userId: '3fa1a820f831422fa7542821bf026080',
          isActive: false,
        },
        {
          emailAddress: 'sushant.jadhav+1234@kibocommerce.com',
          userName: 'sushant.jadhav+1234@kibocommerce.com',
          firstName: 'Sushant',
          lastName: 'Jadhav',
          localeCode: null,
          userId: '86e610a4b16448198b87c0e5d7498ea3',
          isActive: true,
        },
        {
          emailAddress: 'rohit.hedda@gmail.com',
          userName: 'rohit.hedda@gmail.com',
          firstName: 'Rohit',
          lastName: 'Hedda',
          localeCode: 'en-US',
          userId: 'a4eb86cae66a48dc9302942088c869fd',
          isActive: false,
        },
        {
          emailAddress: 'chandradeepta.laha+1234@kibocommerce.com',
          userName: 'chandradeepta.laha+1234@kibocommerce.com',
          firstName: 'Chandradeepta',
          lastName: 'Laha',
          localeCode: 'en-US',
          userId: 'b3c406da70584410865c22be80d5bcc4',
          isActive: true,
        },
        {
          emailAddress: 'bhupendra.s@gmail.com',
          userName: 'bhupendra.s@gmail.com',
          firstName: 'Bhupendra',
          lastName: 'Singh q',
          localeCode: null,
          userId: 'c10d6b24ec3a44998ae8e8f07c921e31',
          isActive: false,
        },
        {
          emailAddress: 'geetanshu.chhabra+1234@kibocommerce.com',
          userName: 'geetanshu.chhabra+1234@kibocommerce.com',
          firstName: 'geetanshu',
          lastName: 'chhabra',
          localeCode: 'en-US',
          userId: 'c43fd8c426a34e5bae5a7445dee2fa86',
          isActive: true,
        },
        {
          emailAddress: 'satish.chavan@gmail.com',
          userName: 'satish.chavan@gmail.com',
          firstName: 'Satish',
          lastName: 'Chavan',
          localeCode: 'en-US',
          userId: 'c5ef0636aa74494182a9ab20590e45fa',
          isActive: false,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: null,
      id: 1174,
      taxId: '3456764',
      accountType: 'B2B',
      companyOrOrganization: 'Sushant Root Account',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant2005@gmail.com',
          userName: 'sushant2005@gmail.com',
          firstName: 'sushant',
          lastName: 'jadhav',
          localeCode: 'en-US',
          userId: 'aca44e18a2044723a951cb65882f5c1d',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1174,
      id: 1177,
      taxId: '123456',
      accountType: 'B2B',
      companyOrOrganization: 'Child 1',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant2006@gmail.com',
          userName: 'sushant2006@gmail.com',
          firstName: 'sushant',
          lastName: 'jadhav',
          localeCode: 'en-US',
          userId: '3b9b2fb50a3144ea888b4c9bbf2633ad',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1177,
      id: 1178,
      taxId: '',
      accountType: 'B2B',
      companyOrOrganization: 'Child 2',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child3@gmail.com',
          userName: 'child3@gmail.com',
          firstName: 'Child',
          lastName: '3',
          localeCode: 'en-US',
          userId: '2c14d1cd82c34b688ac3f20ae399585b',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1178,
      id: 1182,
      taxId: '',
      accountType: 'B2B',
      companyOrOrganization: 'Child 3',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child9@gmail.com',
          userName: 'child9@gmail.com',
          firstName: 'Child',
          lastName: '9',
          localeCode: 'en-IN',
          userId: 'a48bc20e064b4276b782868ec27fd789',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1182,
      id: 1204,
      taxId: '8769869',
      accountType: 'B2B',
      companyOrOrganization: 'Child9',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child10@gmail.com',
          userName: 'child10@gmail.com',
          firstName: 'Child',
          lastName: '10',
          localeCode: 'en-IN',
          userId: 'eb9e0219f2714b5fbf334a646a45b3bc',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1204,
      id: 1205,
      taxId: '6896986986',
      accountType: 'B2B',
      companyOrOrganization: 'Child 10',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child@gmail.com',
          userName: 'child@gmail.com',
          firstName: 'Child',
          lastName: '8',
          localeCode: 'en-IN',
          userId: '835d13aac9b64ec0a568a31a7e788c49',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1178,
      id: 1193,
      taxId: '534645756',
      accountType: 'B2B',
      companyOrOrganization: 'Child 8',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child11@gmail.com',
          userName: 'child11@gmail.com',
          firstName: 'Child',
          lastName: '11',
          localeCode: 'en-IN',
          userId: 'c719d002c42f437ba7ca6fae1ab85431',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1193,
      id: 1206,
      taxId: '869979707',
      accountType: 'B2B',
      companyOrOrganization: 'Child 11',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant3@gmail.com',
          userName: 'sushant3@gmail.com',
          firstName: 'Sushant',
          lastName: '3',
          localeCode: 'en-IN',
          userId: 'afd42af1553b4b8991813f650789deed',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1206,
      id: 1190,
      taxId: '123232',
      accountType: 'B2B',
      companyOrOrganization: 'Sushant 3',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child4@gmail.com',
          userName: 'child4@gmail.com',
          firstName: 'Child',
          lastName: '4',
          localeCode: 'en-US',
          userId: '0c2515927c3c49178841d58a3689a90f',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1177,
      id: 1183,
      taxId: '',
      accountType: 'B2B',
      companyOrOrganization: 'Child 4',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child5@gmail.com',
          userName: 'child5@gmail.com',
          firstName: 'CHild',
          lastName: '5',
          localeCode: 'en-IN',
          userId: 'f936a08ab8df42aa8d217939e55ab65d',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1183,
      id: 1186,
      taxId: '2313242',
      accountType: 'B2B',
      companyOrOrganization: 'Child 5',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child7@gmail.com',
          userName: 'child7@gmail.com',
          firstName: 'Child',
          lastName: '7',
          localeCode: 'en-IN',
          userId: 'b5a5214d9671429389483c6a5f7e3651',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1186,
      id: 1188,
      taxId: '2342342',
      accountType: 'B2B',
      companyOrOrganization: 'Child 7',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'child6@gmail.com',
          userName: 'child6@gmail.com',
          firstName: 'Child',
          lastName: '6',
          localeCode: 'en-IN',
          userId: '2ce6902ecc2a4be5943811c3b2cfc12a',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1186,
      id: 1187,
      taxId: '3243543',
      accountType: 'B2B',
      companyOrOrganization: 'Child 6',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant100@gmail.com',
          userName: 'sushant100@gmail.com',
          firstName: 'Jadhav',
          lastName: null,
          localeCode: 'en-IN',
          userId: '833ef57080ff4354becbb96cd25d3ba8',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1174,
      id: 1199,
      taxId: '123456',
      accountType: 'B2B',
      companyOrOrganization: 'Child 101',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant.jadhav+123456@kibocommerce.com',
          userName: 'sushant.jadhav+123456@kibocommerce.com',
          firstName: 'sushant',
          lastName: 'jadhav',
          localeCode: 'en-US',
          userId: '02c5cd8f7a2642c9ad9c93259625f0b4',
          isActive: true,
        },
        {
          emailAddress: 'ramesh@kibocommerce.com',
          userName: 'ramesh@kibocommerce.com',
          firstName: 'Sushant',
          lastName: 'Jadhav',
          localeCode: 'en-US',
          userId: '477cbd526e694882abc01a457a210a35',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1174,
      id: 1198,
      taxId: null,
      accountType: 'B2B',
      companyOrOrganization: 'Sushant Child Account',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'bhupendrasingh4810@gmail.com',
          userName: 'bhupendrasingh4810@gmail.com',
          firstName: 'Bhupendra',
          lastName: 'Singh',
          localeCode: 'en-IN',
          userId: '419e58dffa754140bc3fee3dafb092c1',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1198,
      id: 1179,
      taxId: '123456',
      accountType: 'B2B',
      companyOrOrganization: 'ABC Enterprise',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'abc1@gmail.com',
          userName: 'abc1@gmail.com',
          firstName: 'ABC',
          lastName: '1',
          localeCode: 'en-US',
          userId: 'bdc2dc04edf14abe9f6dad49426f2da7',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1179,
      id: 1184,
      taxId: '',
      accountType: 'B2B',
      companyOrOrganization: 'ABC 1',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'abc.5@gmail.com',
          userName: 'abc.5@gmail.com',
          firstName: 'ABC',
          lastName: '5',
          localeCode: 'en-IN',
          userId: 'dbd2c077d89940ae9dac1c390a2c8b8f',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1184,
      id: 1194,
      taxId: '655789',
      accountType: 'B2B',
      companyOrOrganization: 'ABC5',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'abc5@gmail.com',
          userName: 'abc5@gmail.com',
          firstName: 'ABC',
          lastName: '4',
          localeCode: 'en-IN',
          userId: '0cccd6e8ca3c4a918e1c45cb5118e1cb',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1194,
      id: 1192,
      taxId: '234567',
      accountType: 'B2B',
      companyOrOrganization: 'ABC4',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'abc12@gmail.com',
          userName: 'abc12@gmail.com',
          firstName: 'ABC',
          lastName: '2',
          localeCode: 'en-IN',
          userId: '98ed25f864cf46fba11b625503cf8ce9',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1179,
      id: 1189,
      taxId: '234232',
      accountType: 'B2B',
      companyOrOrganization: 'ABC2',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'abc4@gmail.com',
          userName: 'abc4@gmail.com',
          firstName: 'ABC',
          lastName: '3',
          localeCode: 'en-IN',
          userId: '05226d511e714800bc870d4548b87596',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1179,
      id: 1191,
      taxId: '123456',
      accountType: 'B2B',
      companyOrOrganization: 'ABC3',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant2@gmail.com',
          userName: 'sushant2@gmail.com',
          firstName: 'shushant',
          lastName: '2',
          localeCode: 'en-US',
          userId: '9d46d8d571a94e07b2ec8f029ebf72d5',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1191,
      id: 1185,
      taxId: '',
      accountType: 'B2B',
      companyOrOrganization: 'Sushant 2',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant101@gmail.com',
          userName: 'sushant101@gmail.com',
          firstName: 'Jadhav',
          lastName: null,
          localeCode: 'en-IN',
          userId: '48c4b3982b8e473692bab2641fa02c1b',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1174,
      id: 1200,
      taxId: '123456',
      accountType: 'B2B',
      companyOrOrganization: 'Child 101',
      contacts: null,
    },
    {
      users: [
        {
          emailAddress: 'sushant.j@gmail.com',
          userName: 'sushant.j@gmail.com',
          firstName: 'Sushant',
          lastName: 'J',
          localeCode: 'en-IN',
          userId: 'd25a88353c8842eebfaacfa5e56d75d1',
          isActive: true,
        },
      ],
      isActive: true,
      rootAccountId: 1174,
      parentAccountId: 1174,
      id: 1196,
      taxId: '656545323',
      accountType: 'B2B',
      companyOrOrganization: 'Sushant',
      contacts: null,
    },
  ],
}

export const hierarchyTreeMock = [
  {
    id: 1174,
    collapsed: false,
    children: [
      {
        id: 1177,
        collapsed: false,
        children: [
          {
            id: 1178,
            collapsed: false,
            children: [
              {
                id: 1182,
                collapsed: false,
                children: [
                  {
                    id: 1204,
                    collapsed: false,
                    children: [
                      {
                        id: 1205,
                        collapsed: false,
                        children: [],
                      },
                    ],
                  },
                ],
              },
              {
                id: 1193,
                collapsed: false,
                children: [
                  {
                    id: 1206,
                    collapsed: false,
                    children: [
                      {
                        id: 1190,
                        collapsed: false,
                        children: [],
                      },
                    ],
                  },
                ],
              },
            ],
          },
          {
            id: 1183,
            collapsed: false,
            children: [
              {
                id: 1186,
                collapsed: false,
                children: [
                  {
                    id: 1188,
                    collapsed: false,
                    children: [],
                  },
                  {
                    id: 1187,
                    collapsed: false,
                    children: [],
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: 1199,
        collapsed: false,
        children: [],
      },
      {
        id: 1198,
        collapsed: false,
        children: [
          {
            id: 1179,
            collapsed: false,
            children: [
              {
                id: 1184,
                collapsed: false,
                children: [
                  {
                    id: 1194,
                    collapsed: false,
                    children: [
                      {
                        id: 1192,
                        collapsed: false,
                        children: [],
                      },
                    ],
                  },
                ],
              },
              {
                id: 1189,
                collapsed: false,
                children: [],
              },
              {
                id: 1191,
                collapsed: false,
                children: [
                  {
                    id: 1185,
                    collapsed: false,
                    children: [],
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: 1200,
        collapsed: false,
        children: [],
      },
      {
        id: 1196,
        collapsed: false,
        children: [],
      },
    ],
  },
]
