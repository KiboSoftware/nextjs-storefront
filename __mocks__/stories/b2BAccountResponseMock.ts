import { B2BAccount } from '@/lib/gql/types'

export const b2BAccountResponseMock: B2BAccount = {
  id: 0,
  parentAccountId: 1023,
  taxId: '123234',
  companyOrOrganization: 'Ignitiv Corp',
}
