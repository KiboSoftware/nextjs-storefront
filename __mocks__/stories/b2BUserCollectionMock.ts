import { B2BUserCollection } from '@/lib/gql/types'

export const customerB2BUserForPage0Mock: B2BUserCollection = {
  items: [
    {
      emailAddress: 'kushagra.agarwal@outlook.com',
      firstName: 'Kushagra',
      lastName: 'Agarwal',
      isActive: true,
      roles: [
        {
          roleId: 1,
          roleName: 'Admin',
        },
      ],
      userId: 'db9c337e8fdf4304b9b3482f4bd3e321',
    },
    {
      emailAddress: 'james.smith@gmail.com',
      firstName: 'James',
      lastName: 'Smith',
      isActive: false,
      roles: [],
      userId: 'db9c337e8fdf4304b9b3482f4bd3e322',
    },
    {
      emailAddress: 'maria.garcia@gmail.com',
      firstName: 'Maria',
      lastName: 'Garcia',
      isActive: true,
      roles: [],
      userId: 'db9c337e8fdf4304b9b3482f4bd3e323',
    },
    {
      emailAddress: 'david.smith@gmail.com',
      firstName: 'David',
      lastName: 'Smith',
      isActive: true,
      roles: [],
      userId: 'db9c337e8fdf4304b9b3482f4bd3e324',
    },
    {
      emailAddress: 'james.johnson@gmail.com',
      firstName: 'James',
      lastName: 'Johnson',
      isActive: false,
      roles: [],
      userId: 'db9c337e8fdf4304b9b3482f4bd3e325',
    },
  ],
  totalCount: 13,
  startIndex: 0,
  pageSize: 5,
  pageCount: 3,
}

export const customerB2BUserForPage1Mock: B2BUserCollection = {
  items: [
    {
      emailAddress: 'Ramesh.Pawar@com',
      firstName: 'Ramesh',
      lastName: 'Pawar',
      isActive: true,
      userId: '076afe3013df4ee897234bcac93c440c',
      roles: [],
    },
    {
      emailAddress: 'addnew@gmail.com',
      firstName: 'add',
      lastName: 'new',
      isActive: true,
      userId: '346e62ecbc5e45139c490a27e0b00755',
      roles: [
        {
          roleId: 3,
          roleName: 'Nonpurchaser',
          roleTags: [],
          assignedInScope: null,
        },
      ],
    },
    {
      emailAddress: 'bruce@wCorp.com',
      firstName: 'Bruce',
      lastName: 'Wayne',
      isActive: true,
      userId: '3d233f17e9814fbca905a60548ba272e',
      roles: [],
    },
    {
      emailAddress: 'kevin.watts@kibocommerce.com',
      firstName: 'kevin',
      lastName: 'watts',
      isActive: true,
      userId: '44a005ddf8c54059a8a7bd65f2bc011d',
      roles: [
        {
          roleId: 1,
          roleName: 'Admin',
          roleTags: [],
          assignedInScope: null,
        },
      ],
    },
    {
      emailAddress: 'Ashish.Kulkarni@com',
      firstName: 'Ashish',
      lastName: 'Kulkarni',
      isActive: false,
      userId: '5a2e5f0215954cf38c84efd12bcd0d14',
      roles: [],
    },
  ],
  totalCount: 13,
  startIndex: 5,
  pageSize: 5,
  pageCount: 3,
}

export const customerB2BUserForPage2Mock: B2BUserCollection = {
  items: [
    {
      emailAddress: 'divya.jain@live.com',
      firstName: 'Divya',
      lastName: 'Jain',
      isActive: true,
      roles: [
        {
          roleId: 2,
          roleName: 'Purchaser',
        },
      ],
      userId: 'db9c337e8fdf4304b9b3482f4bd3e4r5',
    },
    {
      emailAddress: 'ada@gmail.com',
      firstName: 'Ada',
      lastName: 'Maria',
      isActive: false,
      roles: [],
      userId: 'db9c337e8fdf7894b9b3482f4bd3e322',
    },
    {
      emailAddress: 'elizabeth.brown@gmail.com',
      firstName: 'Elizabeth',
      lastName: 'Brown',
      isActive: true,
      roles: [],
      userId: 'db9c337e8fdf5044b9b3482f4bd3e323',
    },
  ],
  totalCount: 13,
  startIndex: 10,
  pageSize: 5,
  pageCount: 3,
}
