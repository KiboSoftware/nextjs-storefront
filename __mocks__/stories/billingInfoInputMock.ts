import { CrBillingInfoInput } from '@/lib/gql/types'

export const billingInfoInputMock: CrBillingInfoInput = {
  billingContact: {
    email: 'chandradeepta.laha@kibocommerce.com',
    firstName: 'John',
    middleNameOrInitial: null,
    lastNameOrSurname: 'Doe',
    companyOrOrganization: null,
    phoneNumbers: {
      home: '9898495849',
      mobile: null,
      work: null,
    },
    address: {
      address1: 'Lamar Street',
      address2: '23/1',
      address3: null,
      address4: null,
      cityOrTown: 'Austin',
      stateOrProvince: 'TX',
      postalOrZipCode: '87878',
      countryCode: 'US',
      addressType: null,
      isValidated: false,
    },
    id: null,
  },
  card: {
    isCardInfoSaved: false,
    paymentOrCardType: 'VISA',
    expireMonth: 1,
    expireYear: 2026,
    paymentServiceCardId: '91ee65434560404488c382a9295526ae',
    cardNumberPartOrMask: '************1111',
    isUsedRecurring: false,
    isTokenized: true,
  },
  paymentType: 'CreditCard',
  paymentWorkflow: 'Mozu',
  isSameBillingShippingAddress: false,
}
