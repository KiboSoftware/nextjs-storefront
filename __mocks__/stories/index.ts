export * from './orderMock'
export * from './LocationInventoryCollectionMock'
export * from './locationCollectionMock'
export * from './ProductCustomMock'
export * from './billingInfoInputMock'
export * from './cartItemMock'
export * from './cartMock'
export * from './categoryFacetDataMock'
export * from './categoryTreeDataMock'
export * from './homePageResultMock'
export * from './configuredProductMock'
export * from './createOrderPaymentActionMock'
export * from './customerAccountCardsMock'
export * from './facetValueMock'
export * from './fulfillmentOptionsMock'
export * from './getUserAddressesMock'
export * from './imageGalleryDataMock'
export * from './locationCollectionMock'
export * from './orderCollection'
export * from './orderMock'
export * from './productOptionSelectMock'
export * from './productOptionTextBoxMock'
export * from './productSearchResultMock'
export * from './searchSuggestionResultMock'
export * from './shippingRateMock'
export * from './updateOrderBillingInfoMock'
export * from './userAddressMock'
export * from './userMock'
export * from './wishlistMock'
export * from './returnReasonsMock'
export * from './createReturnMock'
export * from './checkoutDestinationsMock'
export * from './checkoutMock'
export * from './orderShipmentMock'
export * from './subscriptionCollectionMock'
export * from './CheckoutGroupRatesMock'
export * from './orderSubscriptionNowMock'
export * from './subscriptionMock'
export * from './updateCustomerAccountContact'
