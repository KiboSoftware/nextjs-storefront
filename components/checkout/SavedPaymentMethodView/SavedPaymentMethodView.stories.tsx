import React from 'react'

import { ComponentStory, ComponentMeta } from '@storybook/react'

import SavedPaymentMethodView from './SavedPaymentMethodView'

export default {
  title: 'Checkout/SavedPaymentMethodView',
  component: SavedPaymentMethodView,
} as ComponentMeta<typeof SavedPaymentMethodView>

const Template: ComponentStory<typeof SavedPaymentMethodView> = (args) => (
  <SavedPaymentMethodView {...args} />
)

export const Common = Template.bind({})

Common.args = {
  cardNumberPart: '***********1111',
  expireMonth: 4,
  expireYear: 2026,
  address1: '1234, My Address',
  address2: '1104',
  cityOrTown: 'Austin',
  postalOrZipCode: '78727',
  stateOrProvince: 'Texas',
}

export const Radio = Template.bind({})

Radio.args = {
  ...Common.args,
  radio: true,
  displayRowDirection: false,
  displayTitle: false,
}
