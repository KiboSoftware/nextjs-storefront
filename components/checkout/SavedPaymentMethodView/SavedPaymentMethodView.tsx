import React from 'react'

import { Box, Typography } from '@mui/material'
import { useTranslation } from 'next-i18next'

import { AddressDetailsView, PaymentCardDetailsView } from '@/components/checkout'

import { Card, CrAddress } from '@/lib/gql/types'

interface SavedPaymentMethodViewProps {
  customerAccountCards: Card
  billingAddress: CrAddress
}

const SavedPaymentMethodView = (props: SavedPaymentMethodViewProps) => {
  const { t } = useTranslation('checkout')
  const { customerAccountCards, billingAddress } = props

  //   Should be fetched from Getters
  const cardNumberLength = customerAccountCards.cardNumberPart?.length as number
  const cardLastFourDigits = customerAccountCards.cardNumberPart?.slice(
    cardNumberLength - 4,
    cardNumberLength
  )
  const expireMonth = customerAccountCards.expireMonth as number
  const expireYear = customerAccountCards.expireYear as number

  const streetAddress = billingAddress.address1 as string
  const apartment = billingAddress.address2 as string
  const city = billingAddress.cityOrTown as string
  const state = billingAddress.stateOrProvince as string
  const zipCode = billingAddress.postalOrZipCode as string

  return (
    <Box width={'100%'} maxWidth={873}>
      <Typography variant="h3" fontWeight={'bold'}>
        {t('payment-information')}
      </Typography>
      <Box display="flex" gap={6} pt={2}>
        <PaymentCardDetailsView
          withoutRadioTitle={t('payment-method')}
          cardLastFourDigits={cardLastFourDigits}
          expireMonth={expireMonth}
          expireYear={expireYear}
        />
        <AddressDetailsView
          withoutRadioTitle={t('billing-address')}
          address2={apartment}
          address1={streetAddress}
          cityOrTown={city}
          stateOrProvince={zipCode}
          postalOrZipCode={state}
        />
      </Box>
    </Box>
  )
}

export default SavedPaymentMethodView
