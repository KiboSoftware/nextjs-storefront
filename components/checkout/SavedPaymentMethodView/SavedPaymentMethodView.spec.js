import React from 'react'

import { composeStories } from '@storybook/testing-react'
import { render, screen } from '@testing-library/react'
import { useTranslation } from 'next-i18next'

import * as stories from './SavedPaymentMethodView.stories' // import all stories from the stories file

const { Common } = composeStories(stories)

const PaymentCardDetailsViewMock = () => <div data-testid="payment-card-details-mock" />
jest.mock(
  '@/components/checkout/PaymentCardDetailsView/PaymentCardDetailsView',
  () => PaymentCardDetailsViewMock
)

const AddressDetailsViewMock = () => <div data-testid="address-details-view-mock" />
jest.mock(
  '@/components/checkout/AddressDetailsView/AddressDetailsView',
  () => AddressDetailsViewMock
)

describe('[component] - SavedPaymentMethodView', () => {
  it('should render the component', () => {
    const { t } = useTranslation('checkout')

    render(<Common {...Common.args} />)

    expect(screen.getByTestId('payment-card-details-mock')).toBeVisible()
    expect(screen.getByTestId('address-details-view-mock')).toBeVisible()
    expect(screen.getByText(t('payment-information'))).toBeVisible()
  })
})
