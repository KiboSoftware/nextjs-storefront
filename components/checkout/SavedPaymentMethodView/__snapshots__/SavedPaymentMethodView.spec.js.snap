// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] - SavedPaymentMethodView should render the component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root css-19qelqq"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1nnwfac-MuiTypography-root"
      >
        payment-information
      </h3>
      <div
        class="MuiBox-root css-1pejmll"
      >
        <div
          class="MuiBox-root css-1fuhjh6"
          data-testid="payment-card-details-view"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1a2mweq-MuiTypography-root"
          >
            payment-method
          </h6>
          <div
            class="MuiBox-root css-1tca5tr"
          >
            <div
              class="MuiBox-root css-1m1jmtv"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzIiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="visa"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-k008qs"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-8voym1-MuiTypography-root"
                >
                  ending
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ojil4y-MuiTypography-root"
                >
                  1111
                </span>
              </div>
              <div
                class="MuiBox-root css-k008qs"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-8voym1-MuiTypography-root"
                >
                  exp
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ojil4y-MuiTypography-root"
                >
                  4/2026
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1fuhjh6"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1a2mweq-MuiTypography-root"
          >
            billing-address
          </h6>
          <div
            class="MuiBox-root css-zoser8"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ojil4y-MuiTypography-root"
            >
              1234, My Address
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ojil4y-MuiTypography-root"
            >
              apartment1104
            </p>
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-mu12yc-MuiTypography-root"
              >
                Austin
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-mu12yc-MuiTypography-root"
              >
                78727
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ojil4y-MuiTypography-root"
              >
                Texas
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
